common/src/auxiliary/mod.rs

```diff
@@ -136,3 +140,118 @@ impl<K1, K2, V> Default for DoubleBTreeMap<K1, K2, V> {
         Self::new()
     }
 }
+
+/// An auxiliary storage wrapper type.
+///
+/// Implements DoubleMapStorage and traits like [`IterableByKeyMap`] for such type automatically.
+pub trait AuxiliaryStorageWrap {
+    type Key1: Ord + Clone;
```

Comment:
storage wrap could be just single map

Reply:
so assocs here seem incorrect to me

Reply:
it should be just across T, where T is stored in thread local

Reply:
resolved in DM, we'll settle on using two traits for different storages